entry/makeExtractor.js

```diff
@@ -19,17 +27,48 @@ export default (stats, publicPath) => {
     ? { crossorigin: 'anonymous' }
     : {};
 
+  let previouslyReturnedJsHeadTags = '';
+  let previouslyReturnedCssHeadTags = '';
+
+  const getJsHeadTags = () =>
+    extractor
+      .getLinkTags(extraScriptTagAttributes)
+      .split('\n')
+      .filter(tag => tag.includes('as="script"'))
+      .join('\n');
+
+  const getCssHeadTags = () => extractor.getStyleTags();
```

Comment:
Feel like the code would be much cleaner if `getJsHeadTags` and `getCssHeadTags` returned an array instead of a string?

Reply:
I kept the current pattern to optimise for the predominant use-case: a single call to `getHeadTags`. By keeping these as strings we don't have to split `getStyleTags` if you only ever call `getHeadTags` once.
It's not hard to implement either way, so if you still think an array is better I'll refactor.

Reply:
I think I prefer what I suggested but that's just my opinion. Will leave it up to you.

Reply:
I wonder if we are making things overly complex by trying to jam more logic into a single method.

Would it be easier to seperate the use cases of returning a concatted string, and partially returning the tags as an array into seperate methods?

Reply:
Good call. Separated out into `flushHeadTags`